src/meshapi/views/helpers.py

```diff
@@ -0,0 +1,128 @@
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, OpenApiResponse, extend_schema, inline_serializer
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from meshapi.models import Install, Node
+from meshapi.permissions import HasDisambiguateNumberPermission
+from meshapi.serializers import NestedKeyObjectRelatedField
+from meshapi.serializers.nested_object_references import InstallNestedRefSerializer
+
+helper_err_response_schema = inline_serializer("ErrorResponse", fields={"detail": serializers.CharField()})
+
+install_serializer_field = InstallNestedRefSerializer()
+node_serializer_field = NestedKeyObjectRelatedField(
+    queryset=Node.objects.all(),
+    additional_keys=("network_number",),
+)
+
+
+# TODO: Fix docs, maybe with custom install serializer?
+class DisambiguateInstallOrNetworkNumber(APIView):
+    permission_classes = [HasDisambiguateNumberPermission]
```

Comment:
What do you think about just letting anyone with read-only role or better use this instead of making a permission? Is there a service that needs least privilege?

Reply:
I actually have a project in the works literally right now that needs just this permission and nothing else. Also @james-otten had a use for this too, with his device provisioning kiosk idea

So I'd guess based on that it's not uncommon